aml/org.testeditor.aml.dsl/src/org/testeditor/aml/dsl/validation/AmlValidator.xtend

```diff
@@ -154,40 +157,54 @@ class AmlValidator extends AbstractAmlValidator {
 		val interactionsExpectingButWithoutStrategy = componentElement.componentElementInteractionTypes.filter [
 			defaultMethod !== null && !defaultMethod.locatorStrategyParameters.empty && locatorStrategy == null
 		]
-		if (elementHasNoStrategy && !interactionsExpectingButWithoutStrategy.empty) {
-			val message = '''Element has interactions ('«interactionsExpectingButWithoutStrategy.map[name].join(', ')»') that require a locator strategy, but none is given.'''
-			error(message, COMPONENT_ELEMENT__LOCATOR_STRATEGY, COMPONENT_ELEMENT__LOCATOR_STRATEGY__MISSING)
-		}
-	}
+		if (elementHasNoStrategy &&
+			!interactionsExpectingButWithoutStrategy.
+				empty) {
+					val message = '''Element has interactions ('«interactionsExpectingButWithoutStrategy.map[name].join(', ')»') that require a locator strategy, but none is given.'''
+					error(message, COMPONENT_ELEMENT__LOCATOR_STRATEGY, COMPONENT_ELEMENT__LOCATOR_STRATEGY__MISSING)
+				}
+			}
 
-	override checkImports(XImportSection importSection) {
-		// ignore for now
-	}
+			override checkImports(XImportSection importSection) {
+				// ignore for now
+			}
 
-	/** get a set of strings that occur more than once in the list passed */
-	private def Set<String> getStringsOccuringMoreThanOnce(Iterable<String> list) {
-		val stringsUnusedYet = list.toSet
-		val stringsOccuringMoreThanOnce = list.dropWhile [
-			val stringWasUnusedUpToNow = stringsUnusedYet.contains(it)
-			if (stringWasUnusedUpToNow) {
-				stringsUnusedYet.remove(it) // now used, remove from unused => if used a second time it is not dropped
+			@Check
+			def void checkInteractionNameIsUnique(AmlModel amlModel) {
+				val Map<String, InteractionType> interactionNames = newHashMap
+
+				amlModel.interactionTypes.forEach [
+					if (interactionNames.containsKey(it.name)) {
+						val message = MessageFormat.format(Validation_InteractionType_Name_Dublicate, it.name)
+						error(message, interactionNames.get(it.name),
```

Comment:
even though this fixes a bug previously present in the code, we should add a test to verify this change, shouldn't we?